module/ingress/ingress.go

```diff
@@ -42,20 +44,36 @@ func New(config Config, e *ingest.Engine, chainID flow.ChainID) *Ingress {
 	if config.MaxMsgSize == 0 {
 		config.MaxMsgSize = grpcutils.DefaultMaxMsgSize
 	}
+
+	// create a GRPC server to serve GRPC clients
+	grpcOpts := []grpc.ServerOption{
+		grpc.MaxRecvMsgSize(config.MaxMsgSize),
+		grpc.MaxSendMsgSize(config.MaxMsgSize),
+	}
+
+	// if rpc metrics is enabled, add the grpc metrics interceptor as a server option
+	if config.RpcMetricsEnabled {
+		grpcOpts = append(grpcOpts, grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor))
+	}
+
+	server := grpc.NewServer(grpcOpts...)
+
 	ingress := &Ingress{
 		unit: engine.NewUnit(),
 		handler: &handler{
 			UnimplementedAccessAPIServer: access.UnimplementedAccessAPIServer{},
 			engine:                       e,
 			chainID:                      chainID,
 		},
-		server: grpc.NewServer(
-			grpc.MaxRecvMsgSize(config.MaxMsgSize),
-			grpc.MaxSendMsgSize(config.MaxMsgSize),
-		),
+		server: server,
 		config: config,
 	}
 
+	if config.RpcMetricsEnabled {
+		grpc_prometheus.EnableHandlingTimeHistogram()
```

Comment:
I assume calling this method twice won't cause any issue.

Reply:
it shouldn't but do you see a case where it could be called twice?